clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -478,14 +488,38 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
     } else if (type->isVectorType()) {
       llvm_unreachable("no vector inc/dec yet");
     } else if (type->isRealFloatingType()) {
-      auto isFloatOrDouble = type->isSpecificBuiltinType(BuiltinType::Float) ||
-                             type->isSpecificBuiltinType(BuiltinType::Double);
-      assert(isFloatOrDouble && "Non-float/double NYI");
-
-      // Create the inc/dec operation.
-      auto kind =
-          (isInc ? mlir::cir::UnaryOpKind::Inc : mlir::cir::UnaryOpKind::Dec);
-      value = buildUnaryOp(E, kind, input);
+      if (type->isHalfType())
```

Comment:
In general, whenever following skeleton, I advice to stay closer as possible structurally. Things missing:
- assert+unimplemented for CGFPOptionsRAII
- `&& !CGF.getContext().getLangOpts().NativeHalfType` here

Reply:
Updated.